ocp_resources/stateful_set.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+from ocp_resources.constants import PROTOCOL_ERROR_EXCEPTION_DICT, TIMEOUT_4MINUTES
+from ocp_resources.logger import get_logger
+from ocp_resources.resource import NamespacedResource
+from ocp_resources.utils import TimeoutSampler
+
+
+LOGGER = get_logger(name=__name__)
+
+
+class StatefulSet(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.APPS
+
+    def scale_replicas(self, replica_count):
+        """
+        Update replicas in StatefulSet.
+        Args:
+            replica_count (int): Number of replicas.
+        Returns:
+            StatefulSet is updated successfully
+        """
+        super().to_dict()
+        self.res.update({"spec": {"replicas": replica_count}})
+
+        LOGGER.info(f"Set StatefulSet replicas: {replica_count}")
+        return self.update(resource_dict=self.res)
+
+    def wait_for_replicas(self, deployed=True, timeout=TIMEOUT_4MINUTES):
```

Comment:
We have this code in other places, I think we can move it to resource.py and reuse it.

Reply:
This one is slightly different than in `Deployment`
adds `current_replicas`